torchvision/datasets/kitti.py

```diff
@@ -0,0 +1,171 @@
+import csv
+import os
+from typing import Any, Callable, List, Optional, Tuple
+from urllib.error import URLError
+
+from PIL import Image
+
+from .utils import download_and_extract_archive
+from .vision import VisionDataset
+
+
+class Kitti(VisionDataset):
+    """`KITTI <http://www.cvlibs.net/datasets/kitti>`_ Dataset.
+
+    Args:
+        root (string): Root directory where images are downloaded to.
+            Expects the following folder structure if download=False:
+
+            .. code::
+
+                <root>
+                    └── Kitti
+                        └─ raw
+                            ├── training
+                            |   ├── image_2
+                            |   └── label_2
+                            └── testing
+                                └── image_2
+        split (string): The dataset split to use. One of {``train``, ``test``}.
+            Defaults to ``train``.
+        transform (callable, optional): A function/transform that  takes in an PIL image
+            and returns a transformed version. E.g, ``transforms.ToTensor``
+        target_transform (callable, optional): A function/transform that takes in the
+            target and transforms it.
+        transforms (callable, optional): A function/transform that takes input sample
+            and its target as entry and returns a transformed version.
+        download (bool, optional): If true, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again.
+
+    """
+
+    mirrors = [
+        "https://s3.eu-central-1.amazonaws.com/avg-kitti/",
+    ]
```

Comment:
@fmassa Are we planning on using mirrors everywhere? If yes, I think we should implement this functionality in `download_url`. Otherwise I don't think we should start out with mirrors if the official server is stable.

Reply:
Perhaps I shouldn't have used the name `mirrors` here. This is the link they email you if you try to download the dataset. I kept it this way in case the link changes in future or there are multiple sources.

Reply:
It is not about the name per se. Supporting multiple sources for the dataset makes the download logic more complex. So we should either use a central solution, i.e. by passing `mirrors` to `download_url` or don't implement it at all. Since we have no indication that this server is unstable for some reason, I would simply store a single URL and only use this. This way you need no looping, no error check, and so on.

Reply:
I don't think we should add a `mirrors` functionality to `download_url` -- it just makes the logic more complicated.